### ISSUE TITLE
Text设置shadows后报错

### ISSUE THREAD

**Issue by tomcatter** [Unity-Technologies/com.unity.uiwidgets#281]
错误如下:
```
AssertionError: Another exception was thrown: $Specified cast is not valid.
(wrapper castclass) System.Object.__castclass_with_cache(object,intptr,intptr)
System.Linq.Enumerable+<CastIterator>d__34`1[TResult].MoveNext () (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
System.Collections.Generic.List`1[T]..ctor (System.Collections.Generic.IEnumerable`1[T] collection) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Linq.Enumerable.ToList[TSource] (System.Collections.Generic.IEnumerable`1[T] source) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
Unity.UIWidgets.painting.TextStyle.getTextStyle (System.Single textScaleFactor) (at Packages/com.unity.uiwidgets/Runtime/painting/text_style.cs:513)
Unity.UIWidgets.painting.TextSpan.build (Unity.UIWidgets.ui.ParagraphBuilder builder, System.Single textScaleFactor, System.Collections.Generic.List`1[T] dimensions) (at Packages/com.unity.uiwidgets/Runtime/painting/text_span.cs:42)
Unity.UIWidgets.painting.TextPainter.layout (System.Single minWidth, System.Single maxWidth) (at Packages/com.unity.uiwidgets/Runtime/painting/text_painter.cs:332)
Unity.UIWidgets.rendering.RenderParagraph._layoutText (System.Single minWidth, System.Single maxWidth) (at Packages/com.unity.uiwidgets/Runtime/rendering/paragraph.cs:788)
Unity.UIWidgets.rendering.RenderParagraph._layoutTextWithConstraints (Unity.UIWidgets.rendering.BoxConstraints constraints) (at Packages/com.unity.uiwidgets/Runtime/rendering/paragraph.cs:800)
Unity.UIWidgets.rendering.RenderParagraph.performLayout () (at Packages/com.unity.uiwidgets/Runtime/rendering/paragraph.cs:653)
Unity.UIWidgets.rendering.RenderObject.layout (Unity.UIWidgets.rendering.Constraints constraints, System.Boolean parentUsesSize) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
UnityEngine.Debug:LogException(Exception)
Unity.UIWidgets.foundation.D:logError(String, Exception) (at Packages/com.unity.uiwidgets/Runtime/foundation/debug.cs:63)
Unity.UIWidgets.foundation.UIWidgetsError:dumpErrorToConsole(UIWidgetsErrorDetails, Boolean) (at Packages/com.unity.uiwidgets/Runtime/foundation/assertions.cs:380)
Unity.UIWidgets.foundation.UIWidgetsError:dumpErrorToConsole(UIWidgetsErrorDetails) (at Packages/com.unity.uiwidgets/Runtime/foundation/assertions.cs:357)
Unity.UIWidgets.foundation.UIWidgetsError:reportError(UIWidgetsErrorDetails) (at Packages/com.unity.uiwidgets/Runtime/foundation/assertions.cs:407)
Unity.UIWidgets.rendering.RenderObject:_debugReportException(String, Exception) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:603)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:954)
Unity.UIWidgets.rendering.RenderFlex:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/flex.cs:426)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderPositionedBox:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/shifted_box.cs:331)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderFlex:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/flex.cs:426)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.MultiChildLayoutDelegate:layoutChild(Object, BoxConstraints) (at Packages/com.unity.uiwidgets/Runtime/rendering/custom_layout.cs:63)
Unity.UIWidgets.material._ScaffoldLayout:performLayout(Size) (at Packages/com.unity.uiwidgets/Runtime/material/scaffold.cs:424)
Unity.UIWidgets.rendering.MultiChildLayoutDelegate:_callPerformLayout(Size, RenderBox) (at Packages/com.unity.uiwidgets/Runtime/rendering/custom_layout.cs:128)
Unity.UIWidgets.rendering.RenderCustomMultiChildLayoutBox:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/custom_layout.cs:265)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering._RenderCustomClip`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.cs:941)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderOffstage:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.cs:2633)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.widgets._RenderTheatre:performLayout() (at Packages/com.unity.uiwidgets/Runtime/widgets/overlay.cs:540)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderOffstage:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.cs:2633)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.widgets._RenderTheatre:performLayout() (at Packages/com.unity.uiwidgets/Runtime/widgets/overlay.cs:540)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderProxyBoxMixinRenderObjectWithChildMixinRenderBox`1:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/proxy_box.mixin.gen.cs:60)
Unity.UIWidgets.rendering.RenderObject:layout(Constraints, Boolean) (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:947)
Unity.UIWidgets.rendering.RenderView:performLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/view.cs:98)
Unity.UIWidgets.rendering.RenderObject:_layoutWithoutResize() (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:831)
Unity.UIWidgets.rendering.PipelineOwner:flushLayout() (at Packages/com.unity.uiwidgets/Runtime/rendering/object.cs:456)
Unity.UIWidgets.rendering.RendererBinding:drawFrame() (at Packages/com.unity.uiwidgets/Runtime/rendering/binding.cs:112)
Unity.UIWidgets.widgets.WidgetsBinding:drawFrame() (at Packages/com.unity.uiwidgets/Runtime/widgets/binding.cs:200)
Unity.UIWidgets.rendering.RendererBinding:_handlePersistentFrameCallback(TimeSpan) (at Packages/com.unity.uiwidgets/Runtime/rendering/binding.cs:80)
Unity.UIWidgets.scheduler.SchedulerBinding:_invokeFrameCallback(FrameCallback, TimeSpan, String) (at Packages/com.unity.uiwidgets/Runtime/scheduler/binding.cs:726)
Unity.UIWidgets.scheduler.SchedulerBinding:handleDrawFrame() (at Packages/com.unity.uiwidgets/Runtime/scheduler/binding.cs:655)
Unity.UIWidgets.scheduler.<>c__DisplayClass69_0:<scheduleWarmUpFrame>b__1() (at Packages/com.unity.uiwidgets/Runtime/scheduler/binding.cs:535)
Unity.UIWidgets.async.<>c__DisplayClass10_0:<_createTimer>b__0(Object) (at Packages/com.unity.uiwidgets/Runtime/async/timer.cs:51)
Unity.UIWidgets.async._Timer:_postTaskForTime(IntPtr) (at Packages/com.unity.uiwidgets/Runtime/async/timer.cs:144)

```

`TextStyle` 中`shadows`类型是 `List<BoxShadow>` 而源码 [`text_style.cs`](https://github.com/Unity-Technologies/com.unity.uiwidgets/blob/master/com.unity.uiwidgets/Runtime/painting/text_style.cs)中第`534`行将`BoxShadow`强转成了`Shadow` ：
```
        public ui.TextStyle getTextStyle(float textScaleFactor = 1.0f) {
            return new ui.TextStyle(
                color: color,
                decoration: decoration,
                decorationColor: decorationColor,
                decorationStyle: decorationStyle,
                decorationThickness: decorationThickness,
                fontWeight: fontWeight,
                fontStyle: fontStyle,
                textBaseline: textBaseline,
                fontFamily: fontFamily,
                fontFamilyFallback: fontFamilyFallback,
                fontSize == null ? null : fontSize * textScaleFactor,
                letterSpacing: letterSpacing,
                wordSpacing: wordSpacing,
                height: height,
                // locale: locale,
                foreground: foreground,
                background: background ?? (backgroundColor != null
                    ? new Paint {color = backgroundColor}
                    : null
                ),
                shadows: shadows?.Cast<Shadow>().ToList(),
                fontFeatures: fontFeatures
            );
```

**Comment by zhuxingwei** [comment]
Hi Thanks for the reporting !

Yes it looks to be a bug for me. We will fix it asap and let you know when the fix lands on master

**Comment by zhuxingwei** [comment]
The issue has been fixed by #284
